### PR TITLE
SC-71 # Single set of AWS resources per project

### PR DESCRIPTION
**Pull request by mymattcarroll**
### Added

-   SC-71: `route` property to `request` argument passed to handlers. Will contain the original `route` property.
-   SC-71: analytics log message to wrapper to allow for metrics and analysis on routes

### Changed

-   SC-71: `bm server serverless` now creates one set of AWS resources (Lambda, API Gateway Endpoint and Log Group) for all routes in a project instead of one set for each route.

### Removed

-   SC-71: `/route` input from `bm server logs /route`. Logs will now be retrieved for all routes in the project.
-   SC-71: timeout override at the route level. All routes will now share the same timeout.

### Code Review Notes

-    **Contains Breaking Changes**
-   Probably easier to code review these one at a time.

### CODE REVIEW

File: test/scripts/wrapper.js [hunk 1]
```diff
@@ -106,21 +109,22 @@ test('handler() should return correct boom response', (t) => {
 
 test('handler() should return 500 status code if route is not found', (t) => {
```
Comment (kizaonline) [inline test/scripts/wrapper.js]: Test description is misleading now that it returns 404

**Review by kizaonline — CHANGES_REQUESTED** [review]
test description no longer matches the actual test

**Review by kizaonline — APPROVED** [review]
👍  LGTM

**Review by kizaonline — APPROVED** [review]
👍  LGTM

File: scripts/wrapper.js [hunk 2]
```diff
@@ -57,11 +57,36 @@ function handler (
     statusCode: number
   }) => void */
 ) /* : Promise<void> */ {
+  const startTime = Date.now()
   const request = normaliseLambdaRequest(event)
   const internalHeaders = {}
   internalHeaders['Content-Type'] = 'application/json'
   const finish = (statusCode, body, customHeaders) => {
     const headers = Object.assign(internalHeaders, customHeaders)
+    const endTime = Date.now()
```
Comment (kizaonline) [inline scripts/wrapper.js:66]: Should we include logging the query string and params/body in analytics?

Comment (mymattcarroll) [reply scripts/wrapper.js:66]: Great question, Ill will ask @samblinkmobile.

Comment (mymattcarroll) [reply scripts/wrapper.js:66]: As discussed in person, we will leave these in for now.

**Review by kizaonline — APPROVED** [review]
👍  LGTM

**Pull request merged**